mono/metadata/class.c

```diff
@@ -8089,6 +8089,11 @@ mono_class_is_variant_compatible (MonoClass *klass, MonoClass *oklass, gboolean
 		MonoClass *param1_class = mono_class_from_mono_type (klass_argv [j]);
 		MonoClass *param2_class = mono_class_from_mono_type (oklass_argv [j]);
 
+		if (param1_class->is_array_special_interface && param1_class->valuetype && param2_class->valuetype) {
```

Comment:
What's is this trying to solve?

Reply:
Slightly updated, but it tries to solve this:
```Failed test(s):

bug-400716.exe
=============== bug-400716.exe.stdout ===============
System.Array.InternalEnumerator<System.Byte> doesn't implement interface System.Collections.Generic.IEnumerator<System.SByte>```